app/graphql/mutations/settings/update.rb

```diff
@@ -1,24 +1,59 @@
 module Mutations
   module Settings
-    class Update < UpdateMutation
+    class Update < BaseMutation
       graphql_name 'UpdateSettingMutation'
       description 'Updates a setting'
 
       # https://github.com/graphql/graphql-spec/blob/master/rfcs/InputUnion.md
+      argument :id, ID, required: false
+      argument :name, String, required: false
       argument :value, String, required: true
 
-      field :setting, Types::Setting, 'Setting', null: true
+      field :errors, [Types::AttributeError], null: false
+      field :setting, Types::Setting, description: 'Setting', null: true
 
-      def assign_attributes(object, params)
-        object.parse_string_value(params[:value])
+      def self.resource_class
+        SettingPresenter
       end
 
-      def self.save_object(resource)
-        return super resource unless resource.errors.any?
-        {
-          result_key => resource,
-          :errors => map_errors_to_path(resource),
-        }
+      def resolve(params)
+        authorize!
+
+        if params[:name]
+          name = params[:name]
+        else
+          _, _, name = Foreman::GlobalId.decode(params[:id])
+        end
+
+        definition = Foreman.settings.find(name)
+        validate_object(definition)
+
+        record = Foreman.settings.set_user_value(definition.name, params[:value])
+        save_object(definition, record)
+      end
+
+      def save_object(definition, record)
+        User.as(context[:current_user]) do
+          errors = if record.save
+                     []
+                   else
+                     map_errors_to_path(record)
+                   end
+          {
+            :setting => definition,
+            :errors => errors,
+          }
+        end
+      end
+
+      private
+
+      def authorize!
+        return if context[:current_user]&.can?(:edit_settings)
+
+        raise GraphQL::ExecutionError.new(
+          _('Unauthorized. You do not have the required permission %s.') % permission_name
+        )
```

Comment:
```
```suggestion
        raise GraphQL::ExecutionError.new(
          _('Unauthorized. You do not have the required permission %s.') % :edit_settings
        )
```
 0237edf6 | NameError (undefined local variable or method `permission_name' for #<Mutations::Settings::Update:0x0000000010865370>):
 0237edf6 |   
 0237edf6 | app/graphql/mutations/settings/update.rb:55:in `authorize!'
 0237edf6 | app/graphql/mutations/settings/update.rb:20:in `resolve'
 0237edf6 | app/controllers/api/graphql_controller.rb:59:in `execute_single_graphql_query'
 0237edf6 | app/controllers/api/graphql_controller.rb:20:in `execute'
 0237edf6 | app/controllers/concerns/foreman/controller/timezone.rb:10:in `set_timezone'
 0237edf6 | app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
 0237edf6 | lib/foreman/middleware/libvirt_connection_cleaner.rb:9:in `call'
 0237edf6 | lib/foreman/middleware/telemetry.rb:10:in `call'
 0237edf6 | lib/foreman/middleware/catch_json_parse_errors.rb:9:in `call'
 0237edf6 | lib/foreman/middleware/logging_context_session.rb:22:in `call'
 0237edf6 | lib/foreman/middleware/logging_context_request.rb:11:in `call'
```

Reply:
Thanks :+1: fixed.